src/util/serverModel.ts

```diff
@@ -41,21 +41,22 @@ export class ServerModel {
      * event to see when the server creation is finished
      * @param path path to the server's root directory
      * @param id unique identifier for the newly created server
+     * @param attributes optional extra server attributes
      * @param timeout timeout in milliseconds
      */
-    async createServerFromPathAsync(path: string, id: string, timeout: number = 2000): Promise<Protocol.Status> {
+    async createServerFromPathAsync(path: string, id: string, attributes?: any, timeout: number = 2000): Promise<Protocol.Status> {
```

Comment:
Please don't use the any type. At least use object if there is no more precise matching available.